server/src/hover.ts

```diff
@@ -0,0 +1,80 @@
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { Match, MatchType, Position } from "./parserClasses";
+import { findPreciseMatchOfTypes } from "./parserSearching";
+import { ParseResults } from "./parsingResults";
+import { Hover } from "vscode-languageserver";
+import { getCycles } from "./cycles";
+import path = require("path");
+
+/**
+ * Returns the hover information for the given position in the document.
+ * Currently it supports cycle call and cycle parameter hover. 
+ * @param position the zero-based position in the document 
```

Comment:
Sind auch andere nützliche Hoverinformationen denkbar? In Visual Studio in der C-Programmierung wird zum Beispiel beim Hovern über einen Funktionsnamen der Returnwert, die Parameter und wenn ein Kommentar drüber steht auch dieser angezeigt. Ist bei uns natürlich ein bisschen schwierige, man könnte höchstens vielleicht einen Kommentar bei einem Unterprogramm anzeigen oder was meinst du?

Reply:
Du beziehst dich jetzt auf selber geschriebene Zyklen/Unterprogramme im Workspace? Da wäre natürlich das mit den Kommentaren denkbar, wobei wir ja keine eindeutige Zuordnung von Kommentar zu Unterprogramm wie in anderen Sprachen haben. Da müssten wir dann halt annehmen, dass wenn direkt davor eine Kommentar steht, der dann eine Unterprogramm-Beschreibung ist. Das würde ich dann aber als eigenes Issue sehen.

Die Parameter von den Zyklen die wir jetzt drin haben, kann ich natürlich auch noch in das Hovern vom Funktions-Namen hinzufügen, die hab ich da momentan glaub gar nicht drin.

Reply:
Ok, können wir so machen. War auch nur eine Anmerkung hier für zukünftige Features.

Aber das mit den Parametern kannst mal versuchen wie es wirkt.

Reply:
Das mit den Parameter sieht gut aus finde ich:

![Screenshot 2024-04-19 011409](https://github.com/isg-stuttgart/vscode-isg-cnc/assets/102800599/b550b3b4-44e8-4768-8d8a-8c204b31c4eb)

Reply:
Wenn ich das gerade so sehe, fällt mir auf, dass du wahrscheinlich nur die deutschen Kommentare anzeigst oder? In der json sind aber auch englische Kommentare dabei oder? Könnte man auch abhängig von der gewählten Sprache anzeigen oder? Ist für hier jetzt okay, aber könnte man dann in einem neuen Issue machen.

Reply:
Ne, das wird schon abhängig von der eingestellten Sprache gewählt.